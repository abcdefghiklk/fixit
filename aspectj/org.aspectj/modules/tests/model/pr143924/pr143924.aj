aspect DeclareAnnotation {
    declare @method : * debit(..) : @Secured(role="supervisor");
}

class BankAccount {

    public void debit(String accId,long amount) {
    }
}

@interface Secured {
    String role();
}
