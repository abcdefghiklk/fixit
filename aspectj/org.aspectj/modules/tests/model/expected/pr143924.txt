=== MODEL STATUS REPORT ========= After a batch build
<root>  [java source file] 
  pr143924.aj  [java source file] TEST_SANDBOX\pr143924.aj:1:
    import declarations  [import reference] 
    DeclareAnnotation  [aspect] TEST_SANDBOX\pr143924.aj:1:
      declare @method: * debit(..) : @Secured(role = "supervisor")  [declare @method] TEST_SANDBOX\pr143924.aj:2:
    BankAccount  [class] TEST_SANDBOX\pr143924.aj:5:
      debit(java.lang.String,long)  [method] TEST_SANDBOX\pr143924.aj:7:
    Secured  [annotation] TEST_SANDBOX\pr143924.aj:11:
=== END OF MODEL REPORT =========
=== RELATIONSHIPS REPORT ========= After a batch build
(targets=1) *pr143924.aj}DeclareAnnotation`declare @method (annotates) *pr143924.aj[BankAccount~debit~QString;~J
(targets=1) *pr143924.aj[BankAccount~debit~QString;~J (annotated by) *pr143924.aj}DeclareAnnotation`declare @method
=== END OF RELATIONSHIPS REPORT ==
=== Properties of the model and relationships map =====
method=1
import reference=1
aspect=1
declare @method=1
class=1
annotation=1
RelationshipMapSize=2
FileMapSize=1
java source file=2
